File: worldedit-core/src/main/java/com/boydti/fawe/database/DBHandler.java
package com.boydti.fawe.database;

import com.boydti.fawe.config.Config;
import com.sk89q.worldedit.world.World;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class DBHandler {

    private final Logger log = LoggerFactory.getLogger(Config.class);

    public final static DBHandler IMP = new DBHandler();

    private Map<World, RollbackDatabase> databases = new ConcurrentHashMap<>(8, 0.9f, 1);

    public RollbackDatabase getDatabase(World world) {
        RollbackDatabase database = databases.get(world);
        if (database != null) {
            return database;
        }
        try {
            database = new RollbackDatabase(world);
            databases.put(world, database);
            return database;
        } catch (Throwable e) {
            log.error("No JDBC driver found!", e);
            return null;
        }
    }

}

File: worldedit-core/src/main/java/com/boydti/fawe/database/DBHandler.kt
package com.boydti.fawe.database

import com.boydti.fawe.config.Config
import com.sk89q.worldedit.world.World
import org.slf4j.LoggerFactory
import java.util.concurrent.ConcurrentHashMap

object DBHandler {
    private val log = LoggerFactory.getLogger(Config::class.java)
    private val databases: MutableMap<String, RollbackDatabase> = ConcurrentHashMap(8, 0.9f, 1)
    fun getDatabase(world: World): RollbackDatabase? {
        val worldName = world.name
        var database = databases[worldName]
        return database
                ?: try {
                    database = RollbackDatabase(world)
                    databases[worldName] = database
                    database
                } catch (e: Throwable) {
                    log.error("No JDBC driver found!\n TODO: Bundle driver with FAWE (or disable database)", e)
                    null
                }
    }

}
